File: src/app/screens/recover.py
import json
import re

transcript_path = r"C:\Users\Venpep office\.gemini\antigravity-ide\brain\97b14cbd-e794-49b9-beb2-f396f629a66d\.system_generated\logs\transcript.jsonl"
output_path = r"c:\Users\Venpep office\Downloads\Mip-main (2)\Mip-main\src\app\screens\DashboardViewerScreen.tsx.recovered"

with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            data = json.loads(line)
            # We want to find a VIEW_FILE action where the path is DashboardViewerScreen.tsx
            # Or where the content contains the full text of the file
            if data.get("type") == "VIEW_FILE" and "DashboardViewerScreen.tsx" in data.get("content", ""):
                content = data.get("content", "")
                if "export default function DashboardViewerScreen" in content:
                    print("Found file content in step_index:", data.get("step_index"))
                    
                    # Clean up the output by removing line numbers "1: ", "2: ", etc.
                    lines = content.split('\n')
                    clean_lines = []
                    started = False
                    for l in lines:
                        if "Total Lines:" in l or "File Path:" in l:
                            continue
                        m = re.match(r'^\d+:\s?(.*)', l)
                        if m:
                            clean_lines.append(m.group(1))
                            started = True
                        elif started:
                            # If it's a truncated or other line, we can just keep it or stop
                            if "truncated" in l.lower() or "The above content does NOT show" in l:
                                continue
                            clean_lines.append(l)
                    
                    recovered_code = '\n'.join(clean_lines)
                    with open(output_path, 'w', encoding='utf-8') as out:
                        out.write(recovered_code)
                    print("Successfully wrote recovered file to:", output_path)
                    break
        except Exception as e:
            pass

File: src/app/screens/recover.js
import fs from 'fs';
import path from 'path';

const transcriptPath = "C:\\Users\\Venpep office\\.gemini\\antigravity-ide\\brain\\97b14cbd-e794-49b9-beb2-f396f629a66d\\.system_generated\\logs\\transcript.jsonl";
const outputPath = "c:\\Users\\Venpep office\\Downloads\\Mip-main (2)\\Mip-main\\src\\app\\screens\\DashboardViewerScreen.tsx.recovered";

try {
  const content = fs.readFileSync(transcriptPath, 'utf8');
  const lines = content.split('\n');
  
  for (const line of lines) {
    if (!line.trim()) continue;
    try {
      const data = JSON.parse(line);
      // We are looking for the original load of DashboardViewerScreen.tsx
      if (data.type === "VIEW_FILE" && data.content && data.content.includes("export default function DashboardViewerScreen")) {
        const fileContent = data.content;
        console.log("Found file content in step:", data.step_index);
        
        // Clean line numbers
        const fileLines = fileContent.split('\n');
        const cleanLines = [];
        let started = false;
        
        for (const fl of fileLines) {
          if (fl.includes("Total Lines:") || fl.includes("File Path:")) continue;
          
          const match = fl.match(/^\d+:\s?(.*)/);
          if (match) {
            cleanLines.push(match[1]);
            started = true;
          } else if (started) {
            if (fl.toLowerCase().includes("truncated") || fl.includes("The above content does NOT show")) {
              continue;
            }
            cleanLines.push(fl);
          }
        }
        
        fs.writeFileSync(outputPath, cleanLines.join('\n'), 'utf8');
        console.log("Successfully wrote recovered file to:", outputPath);
        break;
      }
    } catch (e) {
      // ignore parse errors
    }
  }
} catch (err) {
  console.error("Error reading transcript:", err);
}
